devdoc/interlocked_linux_requirements.md

```diff
@@ -0,0 +1,345 @@
+interlocked linux
+================
+
+## Overview
+
+`interlocked linux` is the Linux-specific implementation of the `interlocked` module. Each function calls the corresponding `atomic` function from `stdatomic.h`.
+
+
+## Exposed API
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_add, volatile int32_t*, addend, int32_t, value);

```

Comment:
>volatile [](start = 46, length = 8)

I think all of the types for the stdatomic functions should have the `_Atomic` qualifier. That's not available on win32 since we don't have C11 atomics. It's possible this means we should have some kind of atomic thing defined to make this correct and cross-platform.

Something like:

```
#ifdef WIN32
#define volatile_atomic volatile
#else
#define volatile_atomic volatile _Atomic
#endif
```